.github/workflows/nlp-timeline-gen-workflow.yml

```diff
@@ -0,0 +1,39 @@
+name: nlp-timeline-gen-workflow
+on: push
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.x'
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install flake8 pytest
+          pip install -r requirements.txt
+      - name: Lint with flake8
+        run: |            
+          # stop the build if there are Python syntax errors or undefined names
+          flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+          # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
+          flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
```

Comment:
Would it be possible to put this in a shell script in a `scripts` directory then call the shell script to allow the checks to be run locally aswell? Although in this case it may actually want to be split into separate lint and build steps as you may not always want to do them together.

Reply:
Done.